async/src/main/java/org/odk/collect/async/TaskSpec.kt

```diff
@@ -20,5 +20,6 @@ interface TaskSpec {
 
     companion object {
         const val DATA_PROJECT_ID = "projectId"
+        const val LAST_UNIQUE_ATTEMPT = "lastAttempt"
```

Comment:
This could potentially get passed as a param on `TaskSpec#getTask` rather than as part of the `inputData` leaving that to the extending classes. I'm not sure though. What do you think?

Reply:
`TaskSpec#getTask` has `inputData: Map<String, String>` so using it was a natural choice to me. The same question you could ask about `projectId` - why not to pass it as a parameter?

Reply:
Agreed it could go in either. `projectId` is different as not every `TaskSpec` would (or should) care about Projects (also mentioned [here](https://github.com/getodk/collect/pull/5024#discussion_r805332832)). I do think it'd be better as a parameter as it gives you the simplicity of knowing for sure that the value is present over it being in a map which introduces uncertainty (complexity). I guess which comparing these kinds of things (which feel pretty equal) I'd try to go with more certainty over less, even it's pretty minor.

Reply:
Ok, done.